assets/arc-0031/server/src/types.ts

```diff
@@ -0,0 +1,7 @@
+import type { AlgorandChainId } from 'arc31'
+
+export interface User {
+  authAcc: string
+  challenge: string
+  chainId: AlgorandChainId
```

Comment:
As before, the chainId should be the genesis hash here

Reply:
Addressed in https://github.com/deanstef/ARCs/pull/4/commits/057fa7aa2cc37f11e5e0c4713e8a10f7f8bcd93f

Reply:
We are using `chainId` to refer to the Algorand chain newtorks (MainNet, TestNet and BetaNet) but in the `AuthMessage` interfece we use it to refer to the chain network genesis hash, this is a little bit confusing.

Reply:
Good point. Let's clarify it and agree on the same terminology.

As defined in [ARC-6](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0006.md#string-specification-genesisid-and-genesishash)
- `GenesisID`: is the ASCII string of the Algorand network (e.g. `mainnet-v1.0`)
- `GenesisHash`: is the base64 encoded 32-byte hash of the genesis block

Whereas, as specified in [ARC-25](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0025.md#chain-ids):
- `ChainID`: is a standard numeric identifier of an Algorand chain.

I still think that we MUST use the GenesisID in the AuthMessage. It provides a unique link between the message and an Algorand chain, can be verified, and cannot change.

Let's open an issue and fix that in a different PR (so that we can align also the text)